common4j/src/main/com/microsoft/identity/common/java/crypto/BasicDecryptor.java

```diff
@@ -66,20 +68,73 @@ public byte[] perform() throws ClientException {
         );
     }
 
+    @Override
+    public byte[] decrypt(@NonNull final Key key,
+                          @NonNull final String decryptAlgorithm,
+                          final byte[] iv,
+                          final byte[] dataToBeDecrypted,
+                          final int tagLength,
+                          final byte[] aad) throws ClientException {
+        return performCryptoOperationAndUploadTelemetry(
+                CryptoObjectName.Cipher,
+                decryptAlgorithm,
+                mCryptoFactory,
+                new ICryptoOperation<byte[]>() {
+                    @Override
+                    public byte[] perform() throws ClientException {
+                        return decryptWithCipher(
+                                key, decryptAlgorithm, iv, dataToBeDecrypted, tagLength, aad
+                        );
+                    }
+                }
+        );
+    }
+
     private byte[] decryptWithCipher(@NonNull final Key key,
```

Comment:
decryptWithIv

Reply:
Addressed here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1953/commits/8ef8a06a8f8c110c7e06918e087f97256b0ef69c